src/examples/state_update_rollup.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  Field,
+  state,
+  State,
+  method,
+  PrivateKey,
+  SmartContract,
+  Mina,
+  Party,
+  isReady,
+  Permissions,
+  circuitValue,
+} from 'snarkyjs';
+
+await isReady;
+
+// version of the "simple zkapp" which accepts concurrent updates
+class StateUpdateZkapp extends SmartContract {
+  // the "stateUpdate" field describes a state and how it can be updated
+  // (the state doesn't have to be related to on-chain state. here it is, though)
+  stateUpdate = SmartContract.StateUpdate({
+    // type for the state
+    state: circuitValue<{ counter: Field }>({ counter: Field }),
+    // type for the update
+    update: Field,
+    // function that says how to apply an update
+    apply(state: { counter: Field }, update: Field) {
+      state.counter = state.counter.add(update);
+      return state;
+    },
+  });
+
+  // on-chain version of our state. it will typically lag behind the
+  // version that's implicitly represented by the list of updates
+  @state(Field) counter = State<Field>();
+  // helper field to store the point in the update history that our on-chain state is at
+  @state(Field) stateHash = State<Field>();
```

Comment:
let's change this to whatever you pick on the RFC-side like `actionsHash` or `actionsCommitment`